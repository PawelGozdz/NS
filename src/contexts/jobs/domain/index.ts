export * from './job-position';
export * from './job-user-profile';
export * from './jobs';
export * from './skill';
