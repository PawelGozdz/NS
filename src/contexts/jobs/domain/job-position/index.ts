export * from './errors';
export * from './events';
export * from './job-position-command-repository.interface';
export * from './job-position-entity.fixture';
export * from './job-position-query-repository.interface';
export * from './job-position.entity';
export * from './job-position.snapshot';
