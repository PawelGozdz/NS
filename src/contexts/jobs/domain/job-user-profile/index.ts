export * from './errors';
export * from './events';
export * from './job-user-profile-command-repository.interface';
export * from './job-user-profile-query-repository.interface';
export * from './job-user-profile.aggregate-root';
export * from './job-user-profile.aggregate-root.fixture';
export * from './job-user-profile.snapshot';
