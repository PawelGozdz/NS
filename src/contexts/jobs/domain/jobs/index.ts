export * from './errors';
export * from './events';
export * from './job-command-repository.interface';
export * from './job-query-repository.interface';
export * from './job.aggregate-root';
export * from './job.aggregate-root.fixture';
export * from './job.snapshot';
