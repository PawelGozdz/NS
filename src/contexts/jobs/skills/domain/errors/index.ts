export * from './skill-already-exists.error';
export * from './skill-not-found.error';
