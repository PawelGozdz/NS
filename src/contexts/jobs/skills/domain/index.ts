export * from './errors';
export * from './events';
export * from './skill-command-repository.interface';
export * from './skill-entity.fixture';
export * from './skill-query-repository.interface';
export * from './skill.entity';
