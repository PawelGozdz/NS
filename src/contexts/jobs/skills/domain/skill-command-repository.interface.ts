import { ISkillCreateData, Skill } from './skill.entity';

export abstract class ISkillsCommandRepository {
  abstract save(skill: ISkillCreateData): Promise<{ id: number }>;

  abstract getOneById(id: number): Promise<Skill | undefined>;

  abstract getOneByNameAndCategoryId(name: string, categoryId: number): Promise<Skill | undefined>;
}
