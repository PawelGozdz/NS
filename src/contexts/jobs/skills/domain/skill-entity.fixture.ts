/* eslint-disable @typescript-eslint/strict-boolean-expressions */
import { Skill } from './skill.entity';

type Input = {
  id?: number;
  name?: string;
  description?: string;
  categoryId?: number;
};

export class SkillEntityFixtureFactory {
  public static create(overrides?: Input): Skill {
    const id = overrides?.id ?? 1;
    const name = overrides?.name ? overrides.name : 'Grocery';
    const description = overrides?.description ? overrides.description : 'Grocery Skill';
    const categoryId = overrides?.categoryId ? overrides.categoryId : 3;

    return new Skill({
      id,
      name,
      description,
      categoryId,
    });
  }
}
