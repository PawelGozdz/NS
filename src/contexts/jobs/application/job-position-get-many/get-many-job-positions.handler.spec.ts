import { createMock } from '@golevelup/ts-jest';
import { Test } from '@nestjs/testing';
import { PinoLogger } from 'nestjs-pino';

import { ActorType, IActor } from '@libs/common';

import { IJobPositionQueryRepository, JobPositionInfo } from '../../domain';
import { GetManyJobPositionsHandler } from './get-many-job-positions.handler';
import { GetManyJobPositionsQuery } from './get-many-job-positions.query';

describe('GetManyJobPositionsHandler', () => {
  let handler: GetManyJobPositionsHandler;
  let queryRepositoryMock: jest.Mocked<IJobPositionQueryRepository>;

  beforeEach(async () => {
    queryRepositoryMock = createMock<IJobPositionQueryRepository>();

    const app = await Test.createTestingModule({
      providers: [
        GetManyJobPositionsHandler,
        {
          provide: IJobPositionQueryRepository,
          useValue: queryRepositoryMock,
        },
        {
          provide: PinoLogger,
          useValue: createMock<PinoLogger>(),
        },
      ],
    }).compile();

    handler = app.get(GetManyJobPositionsHandler);
  });

  const actor: IActor = {
    type: ActorType.SYSTEM,
    source: 'source',
  };

  const query: GetManyJobPositionsQuery = {
    _filter: {
      title: 'it',
    },
    actor,
  };

  const categories: JobPositionInfo[] = [
    {
      id: 'position-id',
      title: 'Backend Developer',
      slug: 'backend-developser',
      skillIds: [],
      categoryId: 2,
    },
  ];

  it('should get many job positions', async () => {
    // Arrange
    queryRepositoryMock.getManyBy.mockResolvedValueOnce(categories);

    // Act
    const result = await handler.execute(query);

    // Assert
    expect(result.length).toBe(1);
    expect(result).toMatchSnapshot();
  });
});
