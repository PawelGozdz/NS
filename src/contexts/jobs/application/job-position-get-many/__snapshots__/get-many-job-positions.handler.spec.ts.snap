// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetManyJobPositionsHandler should get many job positions 1`] = `
[
  {
    "categoryId": 2,
    "id": "position-id",
    "skillIds": [],
    "slug": "backend-developser",
    "title": "Backend Developer",
  },
]
`;
